utils/yoke/src/is_covariant.rs

```diff
@@ -0,0 +1,133 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[cfg(feature = "alloc")]
+use alloc::{
+    borrow::{Cow, ToOwned},
+    boxed::Box,
+    rc::Rc,
+};
+
+/// A type implementing `IsCovariant<'a>` is covariant with respect to lifetime `'a`.
+///
+/// All types on which it is safe to implement [`Yokeable`] can also safely
```

Comment:
Not quite, all types _with one lifetime_.

Yokeable allows for there to be other lifetimes which are untouched